core/payments/ondemand/on_demand_validator_metrics.go

```diff
@@ -0,0 +1,107 @@
+package ondemand
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Tracks metrics for the [OnDemandPaymentValidator]
+type OnDemandValidatorMetrics struct {
+	onDemandSymbolCount        prometheus.Histogram
+	onDemandInsufficientFunds  prometheus.Counter
+	onDemandQuorumNotSupported prometheus.Counter
+	onDemandUnexpectedErrors   prometheus.Counter
```

Comment:
do we really need a metrics for this? or is an error log sufficient? Are we expecting this to actually happen more than once in a blue moon? is the purpose here that alerting based on a metrics is simpler than alerting based on log and we don't want to create metrics in grafana based on logs?

Reply:
Weird you seemed to have completely changed your perspective since luke's PR a while ago where he was adding a lot of metrics for each client errors, duplicating logs, and iirc you were also not agreeing with that.

we can add all the metrics we want but at the end of the day someone needs to arrange a diagram to show them, and we need to have alerts setup for them. metrics should be **statistical** averages. They are used to just get an alert/see something went wrong, and then you go digging for what, in the logs.